registration_api.js

```diff
@@ -0,0 +1,694 @@
+            path: this.path,
+            domain: this.domain,
+        });
+    },
+};
+
+
+let sessionState = sessionStorage.getItem('app_registration_state') || 'logged_out';
+const urlParams = new URLSearchParams(window.location.search);
+const token1_in_url = urlParams.get('token1');
+if (token1_in_url) {
+    sessionStorage.setItem('token1', token1_in_url);
+    sessionStorage.setItem('app_registration_state', 'logged_in');
+    sessionState = 'logged_in';
+}
+
+// add getToken function
+const getToken = () => token1_in_url || sessionStorage.getItem('token1');
+
+
+// add setToken function
+const setToken = (token) => {
+    LocalStore.set('config.token', token);
+};
+
+// sest token in localStore if getToken is true
+if (getToken()) {
+    setToken(getToken());
+}
+
+// get app_id from url
+const app_id_in_url = urlParams.get('app_id');
+// if app_id is in url, set it in LocalStore
+if (app_id_in_url) {
+    LocalStore.set('config.app_id', app_id_in_url);
+}
+
+const getSessionAppId = () => app_id_in_url || LocalStore.get("config.app_id");
+
+// get endpoint from url
+const endpoint_in_url = urlParams.get('endpoint');
+// if endpoint is in url, set it as config.server_url
+if (endpoint_in_url) {
+    LocalStore.set('config.server_url', endpoint_in_url);
+}
+
+// add getEndpoint function
+const getEndpoint = () => endpoint_in_url || LocalStore.get('config.server_url');
+
+
+
+const loginUrl = ({ language }) => {
+    const server_url = getEndpoint();
+    const signup_device_cookie = new CookieStorage('signup_device');
+    const signup_device = signup_device_cookie.get('signup_device');
+    const date_first_contact_cookie = new CookieStorage('date_first_contact');
+    const date_first_contact = date_first_contact_cookie.get('date_first_contact');
+    const marketing_queries = `${signup_device ? `&signup_device=${signup_device}` : ''}${date_first_contact ? `&date_first_contact=${date_first_contact}` : ''
+        }`;
+    const getOAuthUrl = () => {
+        return `https://oauth.deriv.com/oauth2/authorize?app_id=${getSessionAppId()}&l=${language}${marketing_queries}&brand=deriv`;
+    };
+
+    if (server_url && /qa/.test(server_url)) {
+        return `https://${server_url}/oauth2/authorize?app_id=${getSessionAppId()}&l=${language}${marketing_queries}&brand=deriv`;
+    }
+    return getOAuthUrl();
+};
+
+function activate(state) {
+    const joinedState = state.toStrings().join(' ');
+    const elApp = document.getElementById('app-registration-machine');
+    const form_checkbox = document.getElementById('expand_form');
+    const is_folded_state = joinedState === 'logged_in logged_in.register_tab logged_in.register_tab.folded_form';
+    if (elApp) {
+        elApp.dataset.state = joinedState;
+        elApp.setAttribute("data-state", joinedState);
+    }
+    if (is_folded_state) {
+        if (form_checkbox) form_checkbox.checked = false;
+    } else {
+        if (form_checkbox) form_checkbox.checked = true;
+    }
```

Comment:
```suggestion
    form_checkbox?.checked = !is_folded_state
```

Reply:
if you want to account for browser compatibility,

`
if (form_checkbox)  form_checkbox.checked = !is_folded_state
`